test/geometry/camera/test_projections.py

```diff
@@ -0,0 +1,225 @@
+
+    @pytest.mark.parametrize("batch_size", [None, 1, 2, 3])
+    def test_cardinality(self, device, dtype, batch_size):
+        self._test_cardinality_project_batch(device, dtype, batch_size)
+        self._test_cardinality_unproject_batch(device, dtype, batch_size)
+
+    def test_project_points_z1(self, device, dtype):
+        points = torch.tensor([1.0, 2.0, 3.0], device=device, dtype=dtype)
+        expected = torch.tensor([0.3333333432674408, 0.6666666865348816], device=device, dtype=dtype)
+        self.assert_close(project_points_z1(points), expected)
+
+    def test_project_points_z1_batch(self, device, dtype):
+        points = torch.tensor([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]], device=device, dtype=dtype)
+        expected = torch.tensor(
+            [
+                [0.3333333432674408, 0.6666666865348816],
+                [0.6666666865348816, 0.8333333730697632],
+            ],
+            device=device,
+            dtype=dtype,
+        )
+        self.assert_close(project_points_z1(points), expected)
+
+    def test_project_points_z1_invalid(self, device, dtype):
+        # NOTE: this is a corner case where the depth is 0.0 and the point is at infinity
+        #      the projection is not defined and the function returns inf. The second point
+        #      is behind the camera which is not a valid point and the user should handle it.
+        points = torch.tensor([[1.0, 2.0, 0.0], [4.0, 5.0, -1.0]], device=device, dtype=dtype)
+        expected = torch.tensor([[torch.inf, torch.inf], [-4.0, -5.0]], device=device, dtype=dtype)
+        self.assert_close(project_points_z1(points), expected)
+
+    def test_unproject_points_z1(self, device, dtype):
+        points = torch.tensor([1.0, 2.0], device=device, dtype=dtype)
+        expected = torch.tensor([1.0, 2.0, 1.0], device=device, dtype=dtype)
+        self.assert_close(unproject_points_z1(points), expected)
+
+    def test_unproject_points_z1_batch(self, device, dtype):
+        points = torch.tensor([[1.0, 2.0], [3.0, 4.0]], device=device, dtype=dtype)
+        expected = torch.tensor([[1.0, 2.0, 1.0], [3.0, 4.0, 1.0]], device=device, dtype=dtype)
+        self.assert_close(unproject_points_z1(points), expected)
+
+    def test_unproject_points_z1_extension(self, device, dtype):
+        points = torch.tensor([1.0, 2.0], device=device, dtype=dtype)
+        extension = torch.tensor([2.0], device=device, dtype=dtype)
+        expected = torch.tensor([2.0, 4.0, 2.0], device=device, dtype=dtype)
+        self.assert_close(unproject_points_z1(points, extension), expected)
+
+    def test_unproject_points_z1_batch_extension(self, device, dtype):
+        points = torch.tensor([[1.0, 2.0], [3.0, 4.0]], device=device, dtype=dtype)
+        extension = torch.tensor([2.0, 3.0], device=device, dtype=dtype)
+        expected = torch.tensor([[2.0, 4.0, 2.0], [9.0, 12.0, 3.0]], device=device, dtype=dtype)
+        self.assert_close(unproject_points_z1(points, extension), expected)
+
```

Comment:
add project_unproject
```suggestion
    def test_project_unproject_points_z1(self, device, dtype):
        points = torch.tensor([1.0, 2.0, 1.0], device=device, dtype=dtype)
        self.assert_close(unproject_points_z1(project_points_z1(points)), points)
```